### DISCUSSION TITLE
Confused about binarySource=install in Kubernetes

### DISCUSSION THREAD

**Discussion by Roshick** [renovatebot/renovate#17617] [Request Help] [answered]
Greetings,

as we have a wide variety of tools and versions I have recently migrated to the slim image of renovate together with binarySource=install. In our setup, the renovate image is running in a kubernetes (cron-) job which is working quite well so far.

Now we were getting "xyz not prepared" for some of the tools in our pull requests and as I looked into the shell scripts I noticed that appearently some tools need to be installed by the root user (such as python here https://github.com/containerbase/buildpack/blob/ebb1c7df7d1c9de411b8ffd5fc6fce479113ee33/src/usr/local/buildpack/tools/v2/python.sh#L62) which fails as the default user in the container is not the root user.

I'm hesitant to simply run the container as root and wanted to know: How am I supposed to use the feature (securily) in Kubernetes?

**Comment by viceice** [comment, marked as answer]
you only need to run `prepare-tool all` as root

- https://github.com/containerbase/sidecar/blob/2c9b008a8c6f8221e4dd29933ee31c6a6aad6278/Dockerfile#L13

**Reply by Roshick** [reply]
That worked, awesome! Thank you! :)
